api/src/main/java/com/capitalone/dashboard/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import com.capitalone.dashboard.auth.AuthProperties;
+import com.capitalone.dashboard.auth.AuthenticationResultHandler;
+import com.capitalone.dashboard.auth.ldap.LdapLoginRequestFilter;
+import com.capitalone.dashboard.auth.standard.StandardLoginRequestFilter;
+import com.capitalone.dashboard.auth.token.JwtAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@EnableConfigurationProperties
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+	
+	@Autowired
+	private JwtAuthenticationFilter jwtAuthenticationFilter;
+	
+	@Autowired
+	private AuthenticationResultHandler authenticationResultHandler;
+
+	@Autowired
+	private AuthenticationProvider standardAuthenticationProvider;
+	
+	@Autowired
+	private AuthProperties authProperties;
+	
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		http.headers().cacheControl();
+		http.csrf().disable()
+			.authorizeRequests().antMatchers("/appinfo").permitAll()
+								.antMatchers("/registerUser").permitAll()
+								.antMatchers("/login**").permitAll()
+								.antMatchers(HttpMethod.GET, "/**").permitAll()
+								
+								// Temporary solution to allow jenkins plugin to send data to the api
+							    //TODO: Secure with API Key
+								.antMatchers(HttpMethod.POST, "/build").permitAll()
+					            .antMatchers(HttpMethod.POST, "/deploy").permitAll()
+					            .antMatchers(HttpMethod.POST, "/artifact").permitAll()
+					            .antMatchers(HttpMethod.POST, "/quality/test").permitAll()
+					            .antMatchers(HttpMethod.POST, "/quality/static-analysis").permitAll()
+					            
+								.anyRequest().authenticated()
+									.and()
+								.addFilterBefore(standardLoginRequestFilter(), UsernamePasswordAuthenticationFilter.class)
+								.addFilterBefore(ldapLoginRequestFilter(), UsernamePasswordAuthenticationFilter.class)
+								.addFilterBefore(jwtAuthenticationFilter, UsernamePasswordAuthenticationFilter.class)
+								.exceptionHandling().authenticationEntryPoint(new Http401AuthenticationEntryPoint("Authorization"));
+	}
+	
+    @Override
+	protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+		auth.authenticationProvider(standardAuthenticationProvider);
+		
+		String ldapServerUrl = authProperties.getLdapServerUrl();
+		String ldapUserDnPattern = authProperties.getLdapUserDnPattern();
+		if (StringUtils.isNotBlank(ldapServerUrl) && StringUtils.isNotBlank(ldapUserDnPattern)) {
+			auth.ldapAuthentication()
+			.userDnPatterns(ldapUserDnPattern)
+			.contextSource().url(ldapServerUrl);
```

Comment:
It wouldn't be that difficult to use corporate ldap groups for controlling permission levels, as long as that was the only way to control admins for ldap users.  Could add properties to the props file for the ldap groups that should be considered admins, and send them through to a custom ldap authorities populator class.  Were you thinking something like that?  Control the access of "standard users" from the app, and "ldap users" using ldap groups?